accepted/00076-live-patching-eol-data.md

```diff
@@ -0,0 +1,119 @@
+- Feature Name: Live Patching EOL Date Information
+- Start Date: 2021-02-08
+- RFC PR:
+
+# Summary
+[summary]: #summary
+
+Enhance live patching integration by providing EOL information on installed patches.
+
+# Motivation
+[motivation]: #motivation
+
+SUSE only provides live patches for kernels up to 1 year after original release date of that kernel. It means that after one year, the user must reboot and use a new kernel so that they can keep getting live patches for another year.
+
+This feature is about enhancing the live patching experience in Uyuni by implementing the following improvements:
+
+1. Add information about the EOL of kernels in the UI
+2. Warn user when a live kernel is approaching EOL, or past EOL
+
+The existing live patching integration in Uyuni is described in the initial [Live Patching RFC](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00027-sle-live-patching.md).
+
+# Detailed design
+[design]: #detailed-design
+
+## Live patching EOL information
+
+SUSE provides lifecycle information on any product to users via 'zypper lifecycle' plugin. The actual data about a product is provided in a plain CSV file in a supplementary package per product. In case of live patching, this package is called `lifecycle-data-sle-live-patching` and is installed together with the product. Zypper lifecycle plugin reads this data to display human-friendly information to users.
+
+Uyuni shall use this data to provide the following:
+ - Display EOL dates for the current live patch in system details
+ - Display an out-of-support systems list in the home page
+ - Use the notifications system and/or send emails to report live patched systems approaching/past EOL
+ - The notifications/emails will be sent in advance according to a time value configurable from the UI
+
+## Data Retrieval
+
+The lifecycle information for a product is provided with `*.lifecycle` files inside a separate package. Uyuni shall unpack this RPM and read the data from the extracted CSV file during reposync of this product. The data shall be stored in the database per package.
+
+An example SQL query to retrieve the list of RPMs to unpack and the corresponding files to read from:
+
+```sql
+SELECT
+    c.name filename,
+    p.path rpm_path
+FROM
+    rhnPackage p
+    JOIN rhnPackageFile f ON p.id = f.package_id
+    JOIN rhnPackageCapability c ON c.id = f.capability_id
+    JOIN rhnPackageEvr e ON p.evr_id = e.id
+WHERE
+    --Select packages that have '*.lifecycle' files in them
+    c.name LIKE '%.lifecycle' AND
+    e.evr = (
+        --Select latest EVR for a package name
+        SELECT MAX(evr)
+        FROM rhnPackageEvr se
+            JOIN rhnPackage sp ON se.id = sp.evr_id
+        WHERE sp.name_id = p.name_id
+    );
```

Comment:
We should limit this to the channel we synced. Otherwise we call this for all channels again and again and also unpack RPM packages when we already have the data.

Or even better we return the package ID of the lifecycle package and store that ID also in the new table.
With this we can exclude lifecycle data packages from unpacking, when the data were already processed

Reply:
Good point. I've updated the SQL example accordingly.